### PR TITLE
Game endpoint testing

### PR DESCRIPTION
**Pull request by MondayHopscotch**
Partial work for #33. Haven't decided how to validate / store player actions, but some testing is rigged up and waiting for us to do that now.

ALSO: SOME NICE HTTP HELPERS ARE IN THIS PR 🌮

### CODE REVIEW

File: server/pogo/game_messages.go [hunk 1]
```diff
@@ -43,3 +50,9 @@ type GameResultSummaryMsg struct {
 	Winner string
 	// TODO: attach stats and other summary information
 }
+
+// CurrentTurnMsg contains information of what turn a game is on
+type CurrentTurnMsg struct {
+	GameID      string `json:"gameID"`
+	CurrentTurn int    `json:"currentTurn"`
```
[the remark this answers is not in the document]
Comment (Zeroff) [reply server/pogo/game_messages.go]: Yeah, had to do that for the gameStartMessage that you reviewed.

[the remark this answers is not in the document]
Comment (Kenoshen) [reply server/pogo/game_messages.go]: 👍

File: server/routing/game.go [hunk 2]
```diff
@@ -13,27 +18,54 @@ const (
...
+	r.HandleFunc(gameRoute+"/{gameID}/turn/{turn}/player/{playerName}", gr.submitCardsHandler).Methods("PUT")
+	r.HandleFunc(gameRoute+"/{gameID}/turn", gr.getCurrentTurnHandler).Methods("GET")
+	r.HandleFunc(gameRoute+"/{gameID}/turn/{turn}/player/{playerName}", gr.getGameStateHandler).Methods("GET")
 }
 
-// submitCardsHandler handles a player request to submit cards for the given tick
+// submitCardsHandler handles a player request to submit cards for the given turn
 func (gr *GameRoutes) submitCardsHandler(w http.ResponseWriter, r *http.Request) {
-	panic("Not yet implemented")
+	w.Write([]byte("Success, though nothing has been recorded"))
```
Comment (Kenoshen) [inline server/routing/game.go]: It probably wouldn't get called directly in this method, but somewhere in the stack here.  But check out rules.input_rules.go.ApplyInputRules to validate the inputs that are received by a player against the current state.

File: server/routing/game.go [hunk 3]
```diff
@@ -13,27 +18,54 @@ const (
...
+
+	gameUUID, err := uuid.FromString(gameID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("%v is not a valid game ID", gameID), http.StatusBadRequest)
+		return
+	}
+
+	game, err := gr.Services.Game.GetActiveGame(gameUUID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to find game with id '%v'", gameID), http.StatusNotFound)
```
Comment (Kenoshen) [inline server/routing/game.go]: After thinking about it, we might want to reserve StatusNotFound for when there is no matching route.  I think typical web servers use 404 for a non-matching route and possibly a 400 when it is something like a non-matching id.

File: server/routing/game_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	lobby.Players = []string{
+		"Player1",
+		"Player2",
+		"Player3",
+	}
+
+	game := svcs.Game.CreateGame(*lobby)
+	ID := game.ID.String()
+
+	resp, err := get(port, gameRoute+"/"+ID+"/turn", []byte{})
```
Comment (Kenoshen) [inline server/routing/game_test.go]: Much cleaner http call 👍

File: server/routing/game_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+	turnBytes, err := json.Marshal(player1Turn)
+	if !assert.Nil(t, err) {
+		t.Fatal(err)
+	}
+
+	// TODO: Actually play legal cards and test accordingly
+
+	resp, err = put(port,
+		gameRoute+"/"+ID+"/turn/"+strconv.Itoa(currentTurn.CurrentTurn)+"/player/Player1",
```
Comment (Kenoshen) [inline server/routing/game_test.go]: I'm surprised you didn't use some sort of string formatting here.  Is there a golang reason for this or just random preference?

File: server/routing/lobby_test.go [hunk 6]
```diff
@@ -48,46 +38,22 @@ func TestLobbyAPI(t *testing.T) {
...
-	assert.Equal(t, "200 OK", r.Status)
-
 	lobbies = svcs.Lobby.GetLobbiesSnapshot()
 	if !assert.Len(t, lobbies[lobbyID].Players, 1) {
 		t.Fatal("expected 1 player in game lobby")
 	}
 	assert.Equal(t, lobbies[lobbyID].Players[0], "TestPlayer1")
 
 	// Create game from our lobby
-	req, err = http.NewRequest(
```
Comment (Kenoshen) [inline server/routing/lobby_test.go]: Yay! No more hard-to-read http requests!

File: server/routing/game.go [hunk 7]
```diff
@@ -13,27 +18,54 @@ const (
 
 // GameRoutes contains information about routes specific to game interaction
 type GameRoutes struct {
-	Service services.GameService
+	Services *services.MasterList
```
Comment (Kenoshen) [inline server/routing/game.go]: I'm not sure I understand what is happening here?  What is the MasterList vs GameService?

Comment (Zeroff) [reply server/routing/game.go]: MasterList is just a struct that has the GameService and a LobbyService so far

File: server/routing/game_test.go (hunk 5 above)
Comment (MondayHopscotch) [reply server/routing/game_test.go]: I was just being lazy. `fmt.Sprintf("my string with number %v", 10)` is the formatting way to do it.

File: server/routing/game.go (hunk 7 above)
Comment (MondayHopscotch) [reply server/routing/game.go]: Yeah, this used to only hold a *GameService pointer, now it just has the whole *MasterList object (which has Game Service as a member). I was just making it similar to how the other route structs look.

**Pull request merged**